metrotranscom/doorway
# feat: add search filter component to landing page and style it

# Pull Request Template

## Description

- Add the missing search filtering component in the hero of the landing page.
- Generate the search URL based on the filters entered.
- Update the hero so the title is in the outer section of it and can be full width, above the image.

Known issues I will try to address today or next week:
- Once button is selected, background color should be doorway-color-blue-500.
- Max monthly rent search box looks weird. Will merge the style from https://github.com/metrotranscom/doorway/pull/194
- For narrow desktop screens, the hero width looks weird and some components overlap.
- Bedroom buttons should be in one row rather than more than one. Right now there is no way to style buttons within a button group so will work on changing that.

If you see any other issues please point them out.

Desktop: 
![image](https://github.com/metrotranscom/doorway/assets/67125998/65ced616-7947-4f0b-a8ce-0c1bf074b2d0)

Mobile: 
![image](https://github.com/metrotranscom/doorway/assets/67125998/f55a637e-0e65-479a-a339-d054bc98430b)
![image](https://github.com/metrotranscom/doorway/assets/67125998/b30581d9-d923-4328-8bad-f1d092ec587b)


## How Can This Be Tested/Reviewed?

- Bring up the public site.
- In the landing page, verify that you see the search filtering component. Compare it against the mocks.
- Click on "view county map" at the bottom. This should open a modal with the county map image. Close the modal.
- Check/enter as many filters as you want and click on "View listings".
- Verify that search filtering worked as intended. You can also check that the URL reflects the filters you entered.
- Go back to the landing page and resize the screen to see how the component is updated.
- Switch to mobile view and compare the component against the mocks.
